### ISSUE TITLE
Launching remote 2D calculations on Expanse

### ISSUE THREAD

**Issue by sinkovit** [sinkovit/Spatial-ecology#161]
Let's start with the 2D case first since nearly everything we do there will apply to remote 3D jobs. I see four steps, the last three of which I imagine will be managed using the reticulate package and/or HubZero capabilities.

(1) Export data from app

The code available at https://github.com/sinkovit/expanse_mkde/blob/main/export-data.R takes the data that is needed for the 2D mkde calculation, serializes it and writes to a file. This can go before the line `mv.dat <- initializeMovementData(t, x, y, sig2obs = sig2obs, t.max = t.max)` in https://github.com/sinkovit/Spatial-ecology/blob/main/bin/compute.R

(2) Transfer data from gateway to Expanse

(3) Run the mkde calculation using code available at https://github.com/sinkovit/expanse_mkde/blob/main/mkde2d-execution.R

(4) Transfer data from Expanse to gateway

**Comment by monaw** [comment]
hi @sinkovit , what portion of our current code will the Expanse calculation replace?  raster and contour or just the mkde part?

**Comment by sinkovit** [comment]
@monaw Only the mkde calculation needs to be done on Expanse

**Comment by monaw** [comment]
@sinkovit, thanks and to confirm, that means only the 4 lines in calculateRaster2D() starting [here](https://github.com/sinkovit/Spatial-ecology/blob/3725cd6a68f115c7a5315c62a9193c8b6f3ed5f3/bin/compute.R#L211)?

**Comment by sinkovit** [comment]
That is correct. You can also look at the following snippet of code to unpack the data that was exported from the shiny app and to pack the results

https://github.com/sinkovit/expanse_mkde/blob/main/mkde2d-execution.R

**Comment by matthewwangg** [comment]
Looks to me like the Python package in question might be Sim2Ls, have we decided on whether we will be utilizing the syscall or reticulate options?

**Comment by matthewwangg** [comment]
As a note, since we weren't able to discuss during the meeting, I received a reply from ACCESS and was able to get the conda environment on Expanse working, the issue was the need to run the command "export OMP_NUM_THREADS=1" and then restart the process
